File: app/src/main/java/org/jay/launchstarterexample/utils/LogUtils.java
package org.jay.launchstarterexample.utils;

import android.util.Log;
import org.jay.launchstarter.utils.Utils;
import org.jay.launchstarterexample.MyApplication;

import java.util.concurrent.ExecutorService;

public class LogUtils {

    private static ExecutorService sExecutorService;

    public static void setExecutor(ExecutorService executorService){
        sExecutorService = executorService;
    }

    public static final String TAG = "performance";

    public static void i(String msg){
        if(Utils.isMainProcess(MyApplication.getApplication())){
            Log.i(TAG,msg);
        }
        // 异步
        if(sExecutorService != null){
//            sExecutorService.execute();
        }
    }

}

File: app/src/main/java/org/jay/appstarterexample/utils/LogUtils.kt
package org.jay.appstarterexample.utils

import android.util.Log
import org.jay.appstarter.utils.Utils
import org.jay.appstarterexample.MyApplication
import java.util.concurrent.ExecutorService

object LogUtils {

    const val TAG = "performance"

    @JvmStatic
    fun i(msg: String?) {
        if (msg == null) return
        if (Utils.isMainProcess(MyApplication.application)) {
            Log.i(TAG, msg)
        }
    }

}